File: DynamicDbWrapper/Converter/DynamicConverter.cs
using System;
using System.Collections.Generic;
using System.Dynamic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DynamicDbWrapper.Converter
{
    public class DynamicConverter
    {
        public static T Convert<T>(ExpandoObject source)
    where T : class
        {
            IDictionary<string, object> dict = source;

            //var ctor = example.GetType().GetConstructors().Single(c => c.GetParameters().Count() > 0);
            var ctor = typeof(T).GetConstructors().Single(c => c.GetParameters().Count() > 0);

            var parameters = ctor.GetParameters();

            var parameterValues = parameters.Select(p => dict[p.Name]).ToArray();

            return (T)ctor.Invoke(parameterValues);
        }
    }
}

File: ToWer.DbWrapper.VB/Converter/DynamicConverter.vb
Imports System.Dynamic

Namespace Converter

    Public Class DynamicConverter

        Public Shared Function Convert(Of T As Class)(ByVal source As ExpandoObject) As T
            Dim dict As IDictionary(Of String, Object) = source
            Dim ctor = GetType(T).GetConstructors().Single(Function(c) c.GetParameters().Count() > 0)
            Dim parameters = ctor.GetParameters()
            Dim parameterValues = parameters.[Select](Function(p) dict(p.Name)).ToArray()
            Return CType(ctor.Invoke(parameterValues), T)
        End Function
    End Class

End Namespace